### PR TITLE
table: fix unstable TestBeginSleepABA

### LINKED ISSUE #36838: TestBeginSleepABA is unstable
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
[2022-08-02T16:06:33.983Z] [2022/08/03 00:05:45.420 +08:00] [INFO] [ddl_worker.go:562] ["[ddl] finish DDL job"] [worker="worker 3, tp general"] [job="ID:70, Type:alter table cache, State:synced, SchemaState:public, SchemaID:2, TableID:68, RowCount:0, ArgLen:0, start time: 2022-08-03 00:05:45.418 +0800 CST, Err:<nil>, ErrCount:0, SnapshotVersion:0"]
[2022-08-02T16:06:33.984Z] [2022/08/03 00:05:45.420 +08:00] [INFO] [ddl.go:1049] ["[ddl] DDL job is finished"] [jobID=70]
[2022-08-02T16:06:33.984Z] [2022/08/03 00:05:45.420 +08:00] [INFO] [callback.go:121] ["performing DDL change, must reload"]
[2022-08-02T16:06:33.984Z] [2022/08/03 00:05:45.421 +08:00] [INFO] [domain.go:154] ["diff load InfoSchema success"] [currentSchemaVersion=34] [neededSchemaVersion=36] ["start time"=398.818µs] [phyTblIDs="[68,68]"] [actionTypes="[144115188075855872,144115188075855872]"]
[2022-08-02T16:06:33.984Z]     cache_test.go:312: 
[2022-08-02T16:06:33.984Z]         	Error Trace:	/home/jenkins/.tidb/tmp/b7c8fa3cc70b7c7023fdc4f79d50df0e/sandbox/processwrapper-sandbox/10502/execroot/__main__/bazel-out/k8-fastbuild/bin/table/tables/tables_test_/tables_test.runfiles/__main__/table/tables/cache_test.go:312
[2022-08-02T16:06:33.984Z]         	Error:      	Should be true
[2022-08-02T16:06:33.984Z]         	Test:       	TestBeginSleepABA
[2022-08-02T16:06:33.984Z] [2022/08/03 00:05:46.396 +08:00] [INFO] [ddl_workerpool.go:82] ["[ddl] closing workerPool"]
[2022-08-02T16:06:33.984Z] [2022/08/03 00:05:46.396 +08:00] [INFO] [ddl_workerpool.go:82] ["[ddl] closing workerPool"]
[2022-08-02T16:06:33.984Z] [2022/08/03 00:05:46.396 +08:00] [INFO] [ddl_worker.go:163] ["[ddl] DDL worker closed"] [worker="worker 3, tp general"] ["take time"=979ns]
[2022-08-02T16:06:33.984Z] [2022/08/03 00:05:46.396 +08:00] [INFO] [ddl_worker.go:163] ["[ddl] DDL worker closed"] [worker="worker 4, tp add index"] ["take time"=264ns]
[2022-08-02T16:06:33.984Z] [2022/08/03 00:05:46.396 +08:00] [INFO] [delete_range.go:148] ["[ddl] closing delRange"]
[2022-08-02T16:06:33.984Z] [2022/08/03 00:05:46.396 +08:00] [INFO] [session_pool.go:94] ["[ddl] closing sessionPool"]
```

### PR DESCRIPTION
**Pull request by xhebox**
Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: ref #36838 

Problem Summary: Ran for one hour without failiures. To prevent `begin` with a earlier `startTS` than the cached results, execute `begin` multiple times.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

**Comment by xhebox** [comment]
After discussing with @tiancaiamao , I decided to skip the test if it not stable. The unstable test case is indeed a bug of cached table. And we now skip it to ensure the logic of testing is correct.

**Review by tiancaiamao — APPROVED** [review]

File: table/tables/cache_test.go [hunk 1]
```diff
@@ -301,15 +301,12 @@ func TestBeginSleepABA(t *testing.T) {
...
-	cacheUsed = false
-	for i := 0; i < 100; i++ {
-		tk1.MustQuery("select * from aba").Check(testkit.Rows("1 1"))
-		if lastReadFromCache(tk1) {
-			cacheUsed = true
-			break
-		}
+	tk1.MustQuery("select * from aba").Check(testkit.Rows("1 1"))
+	if !lastReadFromCache(tk1) {
+		// should read from cache, but it is not stable
```
Comment (wjhuang2016) [inline table/tables/cache_test.go:306]: Add a TODO?

Comment (xhebox) [reply table/tables/cache_test.go]: Done

**Pull request merged**